rpm/containers-common.spec

```diff
@@ -0,0 +1,222 @@
+Source13: %{github_containers}/storage/%{storage_branch}/docs/containers-storage.conf.5.md
+Source14: %{github_containers}/storage/%{storage_branch}/storage.conf
+
+%description
+This package contains common configuration files and documentation for container
+tools ecosystem, such as Podman, Buildah and Skopeo.
+
+It is required because the most of configuration files and docs come from projects
+which are vendored into Podman, Buildah, Skopeo, etc. but they are not packaged
+separately.
+
+%package extra
+Summary: Extra dependencies for Podman and Buildah
+Requires: %{name} = %{epoch}:%{version}-%{release}
+Requires: container-network-stack
+Requires: oci-runtime
+Conflicts: podman < 5:5.0.0~rc4-1
+Recommends: crun
+Recommends: composefs
+Requires: (crun if fedora-release-identity-server)
+Requires: netavark >= 1.10.3-1
+Suggests: slirp4netns
+Requires: passt
+Requires: iptables
+Requires: nftables
+Recommends: qemu-user-static
+Requires: (qemu-user-static-aarch64 if fedora-release-identity-server)
+Requires: (qemu-user-static-arm if fedora-release-identity-server)
+Requires: (qemu-user-static-x86 if fedora-release-identity-server)
+
+%description extra
+This subpackage will handle dependencies common to Podman and Buildah which are
+not required by Skopeo.
+
+%prep
+%autosetup -Sgit %{name}-%{version_no_tilde}
+cp %{SOURCE1} docs/.
+cp %{SOURCE2} docs/.
+cp %{SOURCE3} docs/.
+cp %{SOURCE4} docs/.
+cp %{SOURCE5} docs/.
+cp %{SOURCE6} docs/.
+cp %{SOURCE7} docs/.
+cp %{SOURCE8} docs/.
+cp %{SOURCE9} .
+cp %{SOURCE10} 000-shortnames.conf
+cp %{SOURCE11} .
+cp %{SOURCE12} policy.json
+cp %{SOURCE13} .
+cp %{SOURCE14} .
+
+# Patch storage.conf
+sed -i -e 's/^driver.*=.*/driver = "overlay"/' -e 's/^mountopt.*=.*/mountopt = "nodev,metacopy=on"/' \
+        -e '/additionalimage.*/a "/usr/lib/containers/storage",' \
+%if %{defined copr_build} || (%{defined fedora} && 0%{?fedora} > 40)
+        -e 's/^pull_options.*=.*/pull_options = {enable_partial_images = \"true\", use_hard_links = \"false\", ostree_repos="", convert_images = "false"}/' \
+        -e 's/# use_composefs.*/use_composefs = "false"/g' \
+%endif
+        storage.conf
+
+# Patch seccomp.json
+[ `grep "keyctl" pkg/seccomp/seccomp.json | wc -l` == 0 ] && sed -i '/\"kill\",/i \
+                                "keyctl",' pkg/seccomp/seccomp.json
+sed -i '/\"socketcall\",/i \
+                                "socket",' pkg/seccomp/seccomp.json
+
+# Patch registries.conf
+sed -i 's/^#.*unqualified-search-registries.*=.*/unqualified-search-registries = ["registry.fedoraproject.org", "registry.access.redhat.com", "docker.io"]/g' \
+        registries.conf
+grep '^short-name-mode="enforcing"' registries.conf && echo -e '\nshort-name-mode="enforcing"' >> registries.conf
+
+# Patch containers.conf
+sed -i -e 's/^#.*log_driver.*=.*/log_driver = "journald"/' \
+    -e 's/^#.*compression_format.*=.*/compression_format = "zstd:chunked"/' \
+    pkg/config/containers.conf
```

Comment:
I think we should review this? Why do we patch our own upstream config files so heavily.
I understand that some of them make sense, i.e. adding fedora registries but other are not so obvious. I know this is juts copying the function out of the rpm of today but I really think these should have a clear explanation WHY they are patched in this way.

Reply:
@rhatdan I lifted these from fedora dist-git. How many of these patches are safe to merge/enable upstream?

Reply:
my only 2 cents on this PR would be that scripts could be handled out of the specfile. that's mostly a personal preference as I don't think there's any recommendation on this.

Reply:
@inknos ack thanks. We've been doing something similar in dist-git so far https://src.fedoraproject.org/rpms/containers-common/blob/rawhide/f/update.sh . Let's check if we can reduce these further.

Reply:
(I know, this is not the topic of the PR, feel free to ignore.)

RPM has a native %patch (%autopatch?) mechanism. It seems to me that doing this with manual `sed` all much less readable, and more risky to do something unexpected without us noticing, and I don’t see any benefit.

I tend to think that patch conflicts and application failures are actually our friends.

Does anyone here immediately know what `-e '/additionalimage.*/a "/usr/lib/containers/storage",'` does without referring to the `storage.conf` being edited?!

---

I also agree that if Fedora downstream is using non-default options, that is quite possible but perhaps non-obvious. So having some documentation about which changes are
- forward-ports where we force rawhide to be more experimental than upstream releases
- backports where upstream is ahead of a stable version in a RPM
- long-term differences
along with reasons, would help future maintainers. Or, well, _current_ maintainers asking “How many of these patches are safe to merge/enable upstream?”.

Especially if we move the location where this is maintained, thus requiring extra knowledge to find the original `git` history.

---

FWIW, non-authoritative, just to start the discussion:
- ` -e '/additionalimage.*/a "/usr/lib/containers/storage",' \` feels like CoreOS/image-based-OS specific, not a good upstream default (but I’m not 100% sure)
- `enable_partial_images` etc., `"zstd:chunked"` are experimental and _not_ upstream-ready at this point (ref . https://github.com/containers/storage/issues/1888 )
- `unqualified-search-registries` are Fedora-specific to keep compatibility with older Fedora users, less secure than upstream default
- I don’t immediately know anything about the others (and I didn’t go looking)

Reply:
> Does anyone here immediately know what `-e '/additionalimage.*/a "/usr/lib/containers/storage",'` does without referring to the `storage.conf` being edited?!

there's storage.conf on line 115. Sorry if the readability is poor currently.

And ack other suggestions too.

Reply:
> RPM has a native %patch (%autopatch?) mechanism. It seems to me that doing this with manual sed is all much less readable, and more risky to do something unexpected without us noticing, and I don’t see any benefit.

Sorry for being late to the discussion. +1 for migrating to spec file patches as much as possible in the future.